### PR TITLE
[AIRFLOW-6324] Hide celery commands when executor is not CeleryExecutor

### PR DESCRIPTION
**Pull request by mik-laj**
If the Celery executor is not used, commands specific to it are still displayed. This is confusing to novice users because they try to run commands that do not work.

In addition, the celery commands have been grouped into one subcommand.

| Old command               | New command                        |
|---------------------------|------------------------------------|
| ``airflow worker``        | ``airflow celery worker``          |
| ``airflow flower``        | ``airflow celery flower``          |

---

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6324
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Review by turbaszek — APPROVED** [review]
I like the change!

**Review by potiuk — APPROVED** [review]
I like it too!

**Comment by potiuk** [comment]
One comment - it has big impact on helm charts etc. used to setup celery - but I think note in  UPDATING.md is enough.

**Review by kaxil — APPROVED** [review]

**Comment by mik-laj** [comment]
Side effects eliminated. Can I have this merged?

File: tests/test_configuration.py [hunk 1]
```diff
@@ -344,6 +345,10 @@ def test_broker_transport_options(self):
         self.assertTrue(isinstance(section_dict['_test_only_float'], float))
         self.assertTrue(isinstance(section_dict['_test_only_string'], str))
 
+    @conf_vars({
+        ("celery", "worker_concurrency"): None,
+        ("celery", "celeryd_concurrency"): None,
```
Comment (mik-laj) [inline tests/test_configuration.py:350]: This eliminates the side effects that appear in this test.

File: tests/test_configuration.py [hunk 2]
```diff
@@ -363,6 +368,11 @@ def test_deprecated_options(self):
             self.assertEqual(conf.getint('celery', 'worker_concurrency'), 99)
             conf.remove_option('celery', 'celeryd_concurrency')
 
+    @conf_vars({
+        ("celery", "result_backend"): None,
+        ("celery", "celery_result_backend"): None,
```
Comment (mik-laj) [inline tests/test_configuration.py:373]: This eliminates the side effects that appear in this test.

**Review by turbaszek — APPROVED** [review]

**Pull request merged**